### PR TITLE
Feat/erc 1654

### PR DESCRIPTION
**Pull request by donequis**

### CODE REVIEW

File: utils/rpc/rpc.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	a, err := abi.JSON(strings.NewReader(methodjson))
+
+	h := crypto.Keccak256([]byte(value))
+	hash := [32]byte{}
+	copy(hash[:], h)
+	addr := common.HexToAddress(address)
+
+	packed, err := a.Pack("isValidSignature", hash, common.FromHex(signature))
+	if err != nil {
+		fmt.Println(err)
```
Comment (donequis) [inline utils/rpc/rpc.go]: ^^^

File: handlers/upload_service.go [hunk 2]
```diff
@@ -111,8 +114,24 @@ func (us *UploadServiceImpl) ProcessUpload(r *UploadRequest) error {
...
 	log.Debugf("Validating signature: %s", m.Signature)
+
+	// ERC 1654 support https://github.com/ethereum/EIPs/issues/1654
+	// We need to validate against a contract address whether this is ok or not?
+	if len(m.Signature) > 66 {
+		signature := m.Signature
+		address := m.PubKey
+		msg := fmt.Sprintf("%s.%d", m.Value, m.Timestamp)
+		valid, err := us.rpc.ValidateDapperSignature(address, msg, signature)
+		if err != nil {
```
Comment (m-sossich) [inline handlers/upload_service.go]: Log the error, just to keep track of the Dapper case

File: config.yml [hunk 3]
```diff
@@ -20,6 +20,9 @@ decentralandApi:
 
 logLevel: 'DEBUG' # Set LOG_LEVEL env variable to overwrite this value
 
+infura:
```
Comment (m-sossich) [inline config.yml]: Maybe a more general name than "infura" for this???

File: utils/rpc/rpc.go (hunk 1 above)
Comment (m-sossich) [inline utils/rpc/rpc.go]: please, remove or log properly

File: utils/rpc/rpc.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	}
+
+	res, err := client.CallContract(context.Background(), callMsg, nil)
+	if err != nil {
+		return false, err
+	}
+
+	magic := res[0:4]
+
+	cmp := bytes.Compare(magic, common.FromHex("0x1626ba7e"))
```
Comment (m-sossich) [inline utils/rpc/rpc.go]: `common.FromHex("0x1626ba7e")`
What is that constant? Could you extract it and name it?  Is it something that might be a config?

File: utils/rpc/rpc.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	if cmp == 0 {
+		return true, nil
+	}
+
+	return false, nil
```
Comment (m-sossich) [inline utils/rpc/rpc.go]: maybe just return
`return bytes.Compare(magic, common.FromHex(theConstant)) == 0, nil`

File: utils/rpc/rpc.go (hunk 1 above)
Comment (m-sossich) [reply utils/rpc/rpc.go]: This is not login properly

File: config.yml (hunk 3 above)
Comment (donequis) [reply config.yml]: Any suggestion? Why we can change for web3 rpc, I feel like nothing is as clear as infura right now

File: utils/rpc/rpc.go (hunk 4 above)
Comment (donequis) [reply utils/rpc/rpc.go]: No reason for config nor anything. It's a magic number to check that an address is authorized.

File: handlers/upload_service.go [hunk 6]
```diff
@@ -43,9 +44,10 @@ type UploadServiceImpl struct {
 	Agent           *metrics.Agent
 	ParcelSizeLimit int64
 	Workdir         string
+	rpc             *rpc.RPC
```
[the remark this answers is not in the document]
Comment (donequis) [reply handlers/upload_service.go]: I think we don't need that

File: handlers/upload_service.go (hunk 2 above)
Comment (donequis) [reply handlers/upload_service.go]: Added logs at the service level

File: config.yml (hunk 3 above)
Comment (m-sossich) [reply config.yml]: maybe rpcConnection

**Pull request merged**